### ISSUE TITLE
Not transferring file to the desired folder ( Keyword) in configuration

### ISSUE THREAD

**Issue by drmetro09** [jsavargas/telegram-downloader-bot#8]

**Comment by drmetro09** [comment]
This problem has come after the update 
Telegram downloader is downloading temp file in /download/incompleted and the moving it to /download which is default location , but if in config file i have mentioned keyword “ Combined” so its still moving it to /downlaod and not in /download/combined as specified in config.in
![IMG_1579](https://github.com/user-attachments/assets/3d239242-a4cf-4b87-a42b-d8df600ac00b)

**Comment by jsavargas** [comment]
Yes, that doesn't work like that, only chars_to_replace is enabled, which is used to eliminate characters, not words, when I finish correcting the other errors, I will add that new functionality

**Comment by drmetro09** [comment]
Please add keyword feature again its very helpful to automatically put download files into folders after this update this stopped working

**Comment by jsavargas** [comment]
Fixed, docker image is being generated. And I will put the replace/remove words in the next version

**Comment by drmetro09** [comment]
Thanks i will check it out

**Comment by drmetro09** [comment]
The speed became very slow before it was 30 to 40 mbps now its just 6 to 8 mbps

**Comment by jsavargas** [comment]
I am sure that this is due to when you download a lot, Telegram limits your downloads, because I hardly use it and in the tests I do not see that it downloads slower

**Comment by jsavargas** [comment]
I am uploading a new version, which allows those words to be eliminated, for specific groups or for everyone. I hope it's what you needed

![image](https://github.com/user-attachments/assets/c29104df-8143-40ca-8de1-56d31fe7771d)

**Comment by drmetro09** [comment]
Yes

**Comment by drmetro09** [comment]
> I am sure that this is due to when you download a lot, Telegram limits your downloads, because I hardly use it and in the tests I do not see that it downloads slower

No actually this was first download of the day to test whether keyword function is working but the speed was very low

**Comment by jsavargas** [comment]
Have you done more tests to see if the speed returned to normal or remains slow?

**Comment by jsavargas** [comment]
Have you tried the latest features, that of eliminating words and that of grouping by keywords?

**Comment by drmetro09** [comment]
> Have you done more tests to see if the speed returned to normal or remains slow?

Speed returned to normal

**Comment by drmetro09** [comment]
> Have you tried the latest features, that of eliminating words and that of grouping by keywords?

It is eliminating words and works like charm , thanks ..also also its goes into folders based on keyword.

**Comment by drmetro09** [comment]
Only question i have is if i send /rename command just while forwarding the file to bot will it rename it even before completing the download ?

**Comment by drmetro09** [comment]
> Only question i have is if i send /rename command just while forwarding the file to bot will it rename it even before completing the download ?

**Comment by jsavargas** [comment]
great, if you have any improvements to apply in the following releases

**Comment by jsavargas** [comment]
> Only question i have is if i send /rename command just while forwarding the file to bot will it rename it even before completing the download ?

I think that if you create the rules while downloading it, at the end it could have the new name, but if you rename the file before finishing the download, it does not have to apply the new name to which file yet

**Comment by drmetro09** [comment]
> > Only question i have is if i send /rename command just while forwarding the file to bot will it rename it even before completing the download ?
> 
> I think that if you create the rules while downloading it, at the end it could have the new name, but if you rename the file before finishing the download, it does not have to apply the new name to which file yet

How to create the rule ?

**Comment by jsavargas** [comment]
such as assigning to a folder by group or by words. I have to review and continue improving some regex rules that I had implemented in telethon

**Comment by drmetro09** [comment]
I didn’t get you , could you please explain

**Comment by jsavargas** [comment]
with the options of downloading to specific folders by words, by extension or by groups, this downloads the files in specific folders that you indicate in the config.ini, all of these are rules and they mean that the files are saved once the download is finished in their respective folders. These are all rules.

**Comment by drmetro09** [comment]
But i want the files to be renamed as soon as i add them /forward to the bot

**Comment by jsavargas** [comment]
They are renamed when they finish downloading, I could make a new function that you give them the name before the download finishes and it leaves it saved there until the download finishes and then enters the name that you have given it. What exactly do you want to do?
What names more or less do you want to give it?

**Comment by drmetro09** [comment]
> They are renamed when they finish downloading, I could make a new function that you give them the name before the download finishes and it leaves it saved there until the download finishes and then enters the name that you have given it. What exactly do you want to do? What names more or less do you want to give it?
Y

> They are renamed when they finish downloading, I could make a new function that you give them the name before the download finishes and it leaves it saved there until the download finishes and then enters the name that you have given it. What exactly do you want to do? What names more or less do you want to give it?

“I could make a new function that you give them the name before the download finishes and it leaves it saved there until the download finishes and then enters the name that you have given it” 
- Yes thats what i want , but the names arent same for every file they will be different every time basically I should be able to give the file any name , no restrictions like that

**Comment by drmetro09** [comment]
Hi @jsavargas

**Comment by maniak262** [comment]
When I forward it I can type text in with the forward.  Can you make the forwarded files go into a folder based on that?

EG, i forward a pic and file and add "Test" before i hit enter to forward.  It then saves the files into a subfolder/folder called test.  Or moves to that folder after 100%
